src/frontend/src/components/public/StatusChangeDropDown/populateModal.tsx

```diff
@@ -22,7 +23,11 @@ const PopulateModal = (
   changeStatusFn: ImodalCB,
 ) => {
   const userRoles = roleCheck();
-  const role = userRoles.roles[0];
+  // a drafter does not have a role, but we also allow the drafter can change a status from
+  // MPO to edit_in_progress. so if no roles then this user is a drafter and assign role as DRAFTER
+  let role;
+  if (!deepEqual(userRoles, {})) role = userRoles.roles[0];
```

Comment:
is using Object.keys(userRoles).length === 0 better here?

Reply:
This looks fine to me 🤷🏼‍♂️